system/include/SSE/emmintrin.h

```diff
@@ -1154,7 +1154,13 @@ _mm_loadu_si128(__m128i const *__p)
 static __inline__ __m128i __attribute__((__always_inline__, __nodebug__))
 _mm_loadu_si32(void const *__p)
 {
-  return (__m128i)wasm_i32x4_make(*(unsigned int*)__p, 0, 0, 0);
+  return (__m128i)wasm_f32x4_replace_lane((v128_t)_mm_setzero_ps(), 0, *(float *)__p);
```

Comment:
I don't think switching to `replace_lane` here should change the codegen at all, since the `wasm_i32x4_make` will already compile down a replace_lane into a zero vector.

Are these intrinsics supposed to make any assumption about alignment? This code assumes that `__p` is aligned correctly, and similarly below.

Reply:
The native (SSE) intrinsics don't make any assumptions about alignment. But injecting alignment of 4 into native code is not a big deal, since alignment is just a hint.

Reply:
It's just a hint for the engine, but [not for LLVM](https://llvm.org/docs/LangRef.html#id206:~:text=loads.-,The%20optional%20constant%20align%20argument%20specifies%20the%20alignment%20of%20the%20operation%20(that%20is%2C%20the%20alignment%20of%20the%20memory%20address).%20A%20value%20of%200%20or%20an%20omitted%20align%20argument%20means%20that%20the%20operation%20has%20the%20ABI%20alignment%20for%20the%20target.%20It%20is%20the%20responsibility%20of%20the%20code%20emitter%20to%20ensure%20that%20the%20alignment%20information%20is%20correct.%20Overestimating%20the%20alignment%20results%20in%20undefined%20behavior.%20Underestimating%20the,alignment%20may%20produce%20less%20efficient%20code.,-An%20alignment%20of%201%20is%20always%20safe.%20The%20maximum%20possible%20alignment%20is), which could potentially perform incorrect optimizations if given incorrect alignment information. The most robust way to work around this undefined behavior can be seen in the wasm_simd128.h intrinsics [here](https://github.com/llvm/llvm-project/blob/master/clang/lib/Headers/wasm_simd128.h#L53-L56).

Reply:
Updated to use UB-free snippet from `wasm_simd128.h`